### PR TITLE
dnsdist: Add a Lua FFI helper to generate proxy protocol payloads

### PR DESCRIPTION
**Pull request by rgacogne**
### Checklist

I have:
- [x] read the [CONTRIBUTING.md](https://github.com/PowerDNS/pdns/blob/master/CONTRIBUTING.md) document
- [x] compiled this code
- [x] tested this code
- [ ] included documentation (including possible behaviour changes)
- [ ] documented the code
- [ ] added or modified regression test(s)
- [ ] added or modified unit test(s)

### CODE REVIEW

**Comment by rgacogne** [comment]
Rebased to fix a conflict.

File: pdns/dnsdistdist/dnsdist-lua-ffi.cc [hunk 1]
```diff
@@ -598,3 +598,33 @@ void setupLuaFFIPerThreadContext(LuaContext& luaCtx)
...
 }
+
+size_t dnsdist_ffi_generate_proxy_protocol_payload(const size_t addrSize, const void* srcAddr, const void* dstAddr, const uint16_t srcPort, const uint16_t dstPort, const bool tcp, const size_t valuesCount, const dnsdist_ffi_proxy_protocol_value* values, void* out, const size_t outSize)
+{
+  ComboAddress src, dst;
+  if (addrSize != sizeof(src.sin4.sin_addr) && addrSize != sizeof(src.sin6.sin6_addr.s6_addr)) {
+    return 0;
+  }
+
+  src = makeComboAddressFromRaw(addrSize == sizeof(src.sin4.sin_addr) ? 4 : 6, reinterpret_cast<const char*>(srcAddr), addrSize);
```
Comment (chbruyand) [inline pdns/dnsdistdist/dnsdist-lua-ffi.cc:609]: Just wondering if we want to handle the potential exception that may be thrown here ?

**Review by chbruyand — APPROVED** [review]
LGTM!

File: pdns/dnsdistdist/dnsdist-lua-ffi.cc (hunk 1 above)
Comment (rgacogne) [reply pdns/dnsdistdist/dnsdist-lua-ffi.cc:609]: We do! Because throwing an exception from the FFI interface does not usually end well.

**Comment by rgacogne** [comment]
Build failure is unrelated.

**Pull request merged**